File: BOJ/BOJ - 11866.py
"""
author : Kwak Daeun
github : https://github.com/daeunkwak

title : 요세푸스 문제 0
description : 구현, 자료구조, 큐
"""

from collections import deque

# N명, K번째 사람 제거
N, K = map(int, input().split())

dq = deque(x for x in range (1, N + 1))

cnt = 1
res = []

while dq :
    if cnt % K == 0 :
        cnt += 1
        res.append(dq[0])
        dq.popleft()
    else :
        cnt += 1
        dq.append(dq[0])
        dq.popleft()

print('<', end='')
for i in range(len(res)-1):
    print(res[i], end=', ')
print(res[-1], end='>')

File: BOJ/BOJ - 11866.java
/**
 * author : https://github.com/daeunkwak
 * date : 2023-01-18
 * title : 요세푸스 문제 0
 * description : 자료구조, 큐
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;


public class BOJ_11866 {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        Queue<Integer> q = new LinkedList<>();

        for (int i = 0; i < N; i++) {
            q.add(i+1);
        }
        System.out.print("<");
        while(N-- > 0) {
            for (int i = 0; i < K; i++) {
                if(i == K - 1) {
                    if(N == 0) System.out.print(q.poll());
                    else System.out.print(q.poll() + ", ");
                }else {
                    q.add(q.poll());
                }
            }
        }
        System.out.print(">");



    }
}
